src/cljs/org/broadinstitute/firecloud_ui/page/workspace/summary/library.cljs

```diff
@@ -0,0 +1,155 @@
+                      pk-str (name property-key)]
+                  (when-not hidden
+                    [:div {}
+                     (style/create-form-label (str title (when required? " (required)")))
+                     (cond enum (style/create-identity-select {:ref pk-str} enum)
+                           required? [input/TextField {:ref pk-str
+                                                       :style {:width "100%"}
+                                                       :placeholder inputHint
+                                                       :predicates [(when required? (input/nonempty pk-str))]}]
+                           :else (style/create-text-field {:ref (name property-key)
+                                                           :style {:width "100%"}
+                                                           :placeholder inputHint}))])))
+              (calculate-display-properties library-schema))]
+        (style/create-validation-error-message (:validation-error @state))]))
+   :on-ok
+   (fn [{:keys [props state refs]}]
+     (let [library-schema (:library-schema props)
+           field-data (doall
+                        (map (fn [property-key]
+                               (let [{:keys [hidden enum]} (get-in library-schema [:properties property-key])
+                                     pk-str (name property-key)
+                                     required? (contains? (:required-props @state) property-key)]
+                                 [property-key
+                                  {:required? required?
+                                   :value (not-empty
+                                            (cond hidden (resolve-hidden property-key (:workspace props))
+                                                  enum (common/get-text refs pk-str)
+                                                  required? (do (input/validate refs pk-str)
+                                                              (input/get-text refs pk-str))
+                                                  :else (common/get-text refs pk-str)))}]))
+                             (:propertyOrder library-schema)))
+           missing-fields (not-empty
+                            (keep (fn [[property-key {:keys [required? value]}]]
+                                    (when (and required? (not value)) property-key))
+                                  field-data))]
+       (if missing-fields
+         (swap! state assoc :validation-error ["Please fill out all required fields"])
+         (let [field-data-map (->> field-data
+                                   (keep (fn [[property-key {:keys [value]}]]
+                                           (when value
+                                             [(name property-key) value])))
+                                   (into {}))]
+           (swap! state dissoc :validation-error)
+           ;; TODO: update properties
+           (utils/cljslog field-data-map)))))})
```

Comment:
As I'm sure you know, these get annoying when you're working on something else and want zero noise while doing your own debugging. If this doesn't get triggered because the code is commented-out anyway, no problem.